app/controllers/locations_controller.rb

```diff
@@ -1,25 +1,29 @@
 class LocationsController < ApplicationController
   def create
-    @location = build_location
+    @profile = current_user
+
+    if save_location?
+      enqueue_geocode_job
 
-    if update_location?
       redirect_to profile_url
     else
-      @profile = current_user
-
       render "profiles/show"
     end
   end
 
   private
 
-  def update_location?
-    @location.update(location_params) &&
-      current_user.update(zipcode: location_params[:zipcode])
+  def enqueue_geocode_job
+    GeocodeJob.perform_later(@location)
+  end
+
+  def save_location?
+    build_location.update(location_params) &&
+      @profile.update(zipcode: location_params[:zipcode])
   end
 
   def build_location
```

Comment:
Is it common to call a method like this `build_#{association}`? In some sense it feels more of like "I don't care if it exists or not, just give me a location associated with this profile" so I would just call the method `location`. Not important though, just my opinion.